### ISSUE TITLE
Integração com o github - feature -discussion

### ISSUE THREAD

**Issue by tgcmv** [UNIVALI-LITE/Portugol-Studio#820]
![image](https://user-images.githubusercontent.com/5677568/84513980-a919b300-aca0-11ea-9984-e8e13c8e2e2e.png)

No visual code temos um botão que mostra uma integração com o git, através desse menu podemos fazer um compare, commit, pull, push, etc.
O que acham de uma integração simplificada entre o portugol studio e o github?

Isso também levanta outra questão, o que acham de alterar o portugol studio para 'abrir' um diretório, mostrando lateralmente os arquivos disponiveis?
![image](https://user-images.githubusercontent.com/5677568/84514158-ebdb8b00-aca0-11ea-9b8a-07eb5f8884ac.png)

**Comment by noschang** [comment]
@tgcmv obrigado pela sugestão. Sua ideia é muito pertinente, mas precisamos pensar nela com carinho e cuidado.

O Portugol Studio se diferencia de outras IDEs justamente por ser simples. Você abre o Portugol Studio, clica em um botão e sai programando.

Se exibirmos a árvore de arquivos e fizermos a integração com o GitHub, deixamos o software mais complexo, porque o aluno iniciante vai ter que assimilar os conceitos de "projeto" e "repositório". Não temos certeza se isso é uma boa ideia.

Dentro da equipe já discutimos a hipótese de fazer uma integração com o GitHub, mas de forma transparante. Funcionaria da seguinte forma:

Para usar o Portuol Studio, no primeiro acesso o aluno teria que criar uma conta. Ao criar essa conta, de forma trasparente seria criada também uma conta/repositório para ele no GitHub. Uma vez criada a conta, ao abrir o Portugol Studio o usuário teria que fazer login antes de poder usar. 

A partir daí, todo código que o usuário escrever é automaticamente versionado no repositório de forma transparente cada vez que ele manda compilar ou salvar.

Vantagens:

- O código fica salvo na nuvem, podendo ser acessado de qualquer lugar, basta logar na conta do Portugol Studio

- O código fica versionado, permitindo ao usuário voltar em qualquer versão do código que ele quiser

- O armazenamento do código em repositório permite criar um banco de dados que pode ser usado para fazer análise estatística em cima dos programas criados e com isso melhorar a ferramenta

Desvantagens:

- Requer a criação de uma conta no primeiro acesso

- Requer conexão com a Internet para que funcione adequadamente

- Pode não se possível de implementar devido às políticas de uso do Github, pois nos termos de uso está especificado que um software não pode agir em nome de um usuário (nesse caso a criação automatizada da conta e do repositório)

**Comment by tgcmv** [comment]
Interessante. Sobre as duas ultimas desvantagens podem ser resolvidas com um botão 'salvar na nuvem' e outro 'baixar da nuvem'. O portugol poderia funcionar offline mas ao clicar em salvar na nuvem deverá estar logado e online.

Perde a vantagem de ser totalmente transparente mas elimina 2 desvantagens.
Ao 'salvar' ou 'baixar' o uso do github ficará evidente, porém o uso do git será de forma transparente (o usuário não precisa saber o que é um commit, pull, push, etc.)

Em caso de conflitos: se for clicado em 'salvar na nuvem' considerar código local, caso contrário considerar código do github e ignorar código local.

**Comment by AdsonEsteves** [comment]
Um aluno de uma das aulas desenvolveu um sistema de ligação do Portugol com o git: https://github.com/CarlosPachecoMelo/Plugin-Git

Não possui muitas caracteristicas, mas é possivel.

**Comment by noschang** [comment]
Que legal @AdsonEsteves, não sabia que existia esse plugin. Bem da hora mesmo

**Comment by AdsonEsteves** [comment]
@noschang 
Sim, se me lembro bem ele consegue mandar o arquivo .por para um repositório pelo menos. Lembro que ele não trabalhava com pastas.
Mas se formos desenvolver um um dia, ele serve como uma boa base.
